test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,228 @@
+        this.corfuStoreSink = corfuStoreSink;
+        openMergedTable();
+        this.tableListForGroupA = tableListForGroupA;
+        this.tableListForLocalTable = tableListForLocalTable;
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+        log.error("Error on listener", throwable);
+        fail("onError for LRTestListener: " + throwable.toString());
+    }
+
+    /**
+     * Invoke when snapshot sync starts
+     */
+    @Override
+    protected void onSnapshotSyncStart() {
+        log.info("Snapshot sync started");
+    }
+
+    /**
+     * Apply addedKeys and deletedKeys to merged table, add keys to
+     * existingKeysInMergedTable from addedKeys and deletedKeys.
+     */
+    @Override
+    protected void onSnapshotSyncComplete() {
+        try (TxnContext tx = corfuStoreSink.txn(NAMESPACE)){
+            existingKeysInMergedTable.addAll(addedKeys.keySet());
+            addedKeys.forEach((addedKey, e) -> {
+                tx.putRecord(mergedTable, (Sample.StringKey) addedKey,
+                        SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString()).build(), e.getMetadata());
+            });
+
+            existingKeysInMergedTable.removeAll(deletedKeys);
+            deletedKeys.forEach(deletedKey -> {
+                tx.delete(mergedTable, (Sample.StringKey) deletedKey);
+            });
+
+            tx.commit();
+        }
+
+        log.info("Snapshot sync complete");
+    }
+
+    /**
+     * Write local tables to merged table, process updates and add the keys to addedKeys
+     * and deletedKeys for replicated table.
+     *
+     * @param results Entries received in a single transaction as part of a snapshot sync
+     */
+    @Override
+    protected void processUpdatesInSnapshotSync(CorfuStreamEntries results) {
+        log.info("Processing updates in snapshot sync");
+        results.getEntries().forEach((schema, entries) -> {
+            entries.forEach(e -> {
+                Message key = e.getKey();
+                if (schema.getTableName().contains("Local_Table00")) {
+                    System.out.println(schema.getTableName());
+                    System.out.println(mergedTable.count());
+                    if (!(e.getOperation() == CorfuStreamEntry.OperationType.CLEAR) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        try (TxnContext tx = corfuStoreSink.txn(namespace)){
+                            if (e.getOperation() == CorfuStreamEntry.OperationType.UPDATE) {
+                                tx.putRecord(mergedTable, (Sample.StringKey) key,
+                                                SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString()).build(),
+                                         e.getMetadata());
+                                existingKeysInMergedTable.add(key);
+                            }
+                            else {
```

Comment:
do you want to check delete operation here?

Reply:
+1, it's definitely needed here

Reply:
Sure. Just added an if clause to check delete operation.